src/streamlink/plugins/pluto.py

```diff
@@ -0,0 +1,113 @@
+    api_url_movies = 'https://api.pluto.tv/v3/vod/categories?includeItems=true&deviceType=web'
+
+    media_schema = {
+        'slug': validate.text,
+        validate.optional('stitched'): {
+            'urls': [{
+                'url': validate.url()
+            }]
+        }
+    }
+
+    live_channels_schema = validate.Schema([
+        media_schema
+    ])
+
+    tv_schema = validate.Schema({
+        'seasons': [{
+            'episodes': [
+                media_schema
+            ]
+        }]
+    })
+
+    movies_schema = validate.Schema({
+        'categories': [{
+            'items': [
+                media_schema
+            ]
+        }]
+    })
+
+    @classmethod
+    def can_handle_url(cls, url):
+        return (
+            re.match(cls.url_re_live, url)
+            or re.match(cls.url_re_tv, url)
+            or re.match(cls.url_re_movie, url)
+        )
+
+    def _get_streams(self):
+        match = None
+
+        # live TV
+        if re.match(self.url_re_live, self.url):
+            slug = re.match(self.url_re_live, self.url).group('slug').lower()
+
+            channels_res = self.session.http.get(self.api_url_live_channels)
+            channels_data = self.session.http.json(channels_res, schema=self.live_channels_schema)
+
+            match = next(filter(lambda channel: channel['slug'] == slug, channels_data), None)
+
+        # On-demand TV
+        elif re.match(self.url_re_tv, self.url):
+            re_match = re.match(self.url_re_tv, self.url)
+            series_slug = re_match.group('series_slug')
+            episode_slug = re_match.group('episode_slug')
+
+            tv_series_res = self.session.http.get(self.api_url_tv_series_template % series_slug)
+            tv_series_data = self.session.http.json(tv_series_res, schema=self.tv_schema)
+
+            for season in tv_series_data['seasons']:
+                for episode in season['episodes']:
+                    if episode['slug'] == episode_slug:
+                        match = episode
+
+        # On-demand movie
+        elif re.match(self.url_re_movie, self.url):
+            slug = re.match(self.url_re_movie, self.url).group('slug').lower()
+
+            movies_res = self.session.http.get(self.api_url_movies)
+            movies_data = self.session.http.json(movies_res, schema=self.movies_schema)
+
+            for category in movies_data['categories']:
+                for movie in category['items']:
+                    if movie['slug'] == slug:
+                        match = movie
+
+        return self.handle_media_match(match)
+
+    def handle_media_match(self, match):
+        if not match:
+            return
+
+        if 'stitched' not in match:
+            log.error('No streams found. This video may use an unsupported format.')
+            return
+
+        stream_link_no_sid = match['stitched']['urls'][0]['url']
+        sid = str(uuid4())
+        stream_link = stream_link_no_sid.replace('&sid=', '&sid=' + sid)
```

Comment:
you could use `update_qsd`

https://github.com/streamlink/streamlink/blob/ad276c4eefae7aea23428ca927952ece7c619f34/src/streamlink/utils/url.py#L66
https://github.com/streamlink/streamlink/blob/master/tests/test_utils_url.py#L45-L48

Reply:
Whenever I try to use `update_qsd`, it removes all the url params with no value, which breaks this plugin. 

So `update_qsd('http://url.com?param1=&param2=', {'param1': '', 'param2': 'value'})` becomes `http://url.com?param2=value`

`{'param1': None}` also gives the same result

Reply:
Sounds like a bug in `update_qsd`...